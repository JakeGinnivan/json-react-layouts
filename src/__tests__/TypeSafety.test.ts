import path from 'path'
import spawn from 'cross-spawn'

test('component-library/__App/component-rendering', () => {
    const typescriptCompilation = spawn.sync('./node_modules/.bin/tsc', [
        '-p',
        path.resolve(__dirname, '../TypeSafetyTestFixtures/testconfig.json'),
    ])

    const output = typescriptCompilation.stdout.toString()

    expect(output).toMatchSnapshot('Typescript expected failures')
})
