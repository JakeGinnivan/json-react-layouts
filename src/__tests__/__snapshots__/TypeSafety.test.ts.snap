// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component-library/__App/component-rendering: Typescript expected failures 1`] = `
"src/TypeSafetyTestFixtures/page-renderer.tsx(24,17): error TS2322: Type 'string' is not assignable to type '\\"test-composition\\"'.
src/TypeSafetyTestFixtures/page-renderer.tsx(39,17): error TS2322: Type 'string' is not assignable to type '\\"test-composition\\"'.
src/TypeSafetyTestFixtures/page-renderer.tsx(41,42): error TS2322: Type 'string' is not assignable to type '\\"nested-composition\\" | \\"test\\"'.
src/TypeSafetyTestFixtures/registration.tsx(13,15): error TS2345: Argument of type '\\"test2\\"' is not assignable to parameter of type '\\"test\\"'.
src/TypeSafetyTestFixtures/route-builder-components.tsx(17,5): error TS2322: Type '\\"missing\\"' is not assignable to type '\\"nested-composition\\" | \\"test\\"'.
src/TypeSafetyTestFixtures/route-builder-components.tsx(23,5): error TS2322: Type '\\"missing\\"' is not assignable to type '\\"nested-composition\\" | \\"test\\"'.
src/TypeSafetyTestFixtures/route-builder-components.tsx(28,28): error TS2322: Type '\\"missing\\"' is not assignable to type '\\"test-composition\\"'.
src/TypeSafetyTestFixtures/route-builder-components.tsx(28,56): error TS2741: Property 'main' is missing in type '{}' but required in type '{ main: (ComponentInformation<\\"nested-composition\\", NestedCompositionProps> | ComponentInformation<\\"test\\", {}>)[]; }'.
"
`;
